### PR TITLE
Package Editor and Compiler should do version checks on keyboards

### PR DESCRIPTION
**Pull request by mcdurdin**
This PR implements version checks for packages, to ensure that (a) multiple compiled keyboards with the same ID are not included in a single package, and (b) that version numbers for a given keyboard match (i.e. when both .kmx and .js for the keyboard are in a single package).

### CODE REVIEW

File: windows/src/developer/TIKE/compile/CompileKeymanWeb.pas [hunk 1]
```diff
@@ -2326,19 +2326,25 @@ function TCompileKeymanWeb.GetKeyboardModifierBitmask: string;
 begin
   FBitMask := 0;
   gp := fk.dpGroupArray;
-  for i := 0 to fk.cxGroupArray-1 do
+  if fk.cxGroupArray > 0 then
```
Comment (mcdurdin) [inline windows/src/developer/TIKE/compile/CompileKeymanWeb.pas]: This was a separate bug (cxGroupArray is unsigned so if no groups/rules in a group, -1 gives integer underflow. I don't want to change the sign of the variable without a lot more validation so this is quick easy fix.

File: windows/src/global/delphi/packages/Keyman.System.PackageInfoRefreshKeyboards.pas [hunk 2]
```diff
@@ -77,9 +109,48 @@ procedure TPackageInfoRefreshKeyboards.Execute;
...
+    ids := FindKeyboardFilesByID(pack.Keyboards[i].ID);
+    Assert(Length(ids) > 0);
+
+    if Length(ids) = 1 then
+      Continue;
+
+    if Length(ids) > 2 then
+    begin
+      DoError(Format(SError_TooManyTargetFilesForKeyboardID, [pack.Keyboards[i].ID, Length(ids)]), plsError);
+      Exit(False);
```
Comment (jahorton) [inline windows/src/global/delphi/packages/Keyman.System.PackageInfoRefreshKeyboards.pas]: I assume, based on this logic, that this is omitting the possibility for a LDML keyboard at this time.  Which is reasonable; I'm just wanting to ensure I understand what this block and the one immediately after it are doing.

File: windows/src/developer/TIKE/child/UfrmPackageEditor.pas [hunk 3]
```diff
@@ -1229,15 +1232,25 @@ procedure TfrmPackageEditor.CreateFromCompiledKeyboard(FKMXFilename, FJSFilename
...
+  frmMessages.Clear;
+
   n := lbKeyboards.ItemIndex;
 
   with TPackageInfoRefreshKeyboards.Create(pack) do
   try
-    Execute;
+    OnError := Self.HandlePackageRefreshError;
+    if not Execute then
+      frmMessages.Show;
```
Comment (jahorton) [inline windows/src/developer/TIKE/child/UfrmPackageEditor.pas]: I don't see what `frmMessages.Show` displays, but I assume it's an error/feedback message about keyboard version issues?  And that the `Execute` here is `TPackageInfoRefreshKeyboards.Execute`?

File: windows/src/global/delphi/packages/Keyman.System.PackageInfoRefreshKeyboards.pas (hunk 2 above)
Comment (mcdurdin) [reply windows/src/global/delphi/packages/Keyman.System.PackageInfoRefreshKeyboards.pas]: Yes, we'll add the LDML logic later :)

File: windows/src/developer/TIKE/child/UfrmPackageEditor.pas (hunk 3 above)
Comment (mcdurdin) [reply windows/src/developer/TIKE/child/UfrmPackageEditor.pas]: `frmMessages` is the error message form which should be shown if there are any errors. And yes, `Execute`.

**Pull request merged**